diff --git a/node_modules/next-auth/react.js b/node_modules/next-auth/react.js
index e667d46..08befc6 100644
--- a/node_modules/next-auth/react.js
+++ b/node_modules/next-auth/react.js
@@ -154,11 +154,11 @@ export async function signIn(provider, options, authorizationParams) {
     // TODO: Do not redirect for Credentials and Email providers by default in next major
     if (redirect || !isSupportingReturn) {
         const url = data.url ?? callbackUrl;
-        window.location.href = url;
+        // window.location.href = url;
         // If url contains a hash, the browser does not reload the page. We reload manually
         if (url.includes("#"))
             window.location.reload();
-        return;
+        return url;
     }
     const error = new URL(data.url).searchParams.get("error");
     const code = new URL(data.url).searchParams.get("code");